cockatrice/src/window_main.cpp

```diff
@@ -86,7 +86,12 @@ void MainWindow::processConnectionClosedEvent(const Event_ConnectionClosed &even
 
 void MainWindow::processServerShutdownEvent(const Event_ServerShutdown &event)
 {
-    QMessageBox::information(this, tr("Scheduled server shutdown"), tr("The server is going to be restarted in %n minute(s).\nAll running games will be lost.\nReason for shutdown: %1", "", event.minutes()).arg(QString::fromStdString(event.reason())));
+    if (serverShutdownMessageBox)
+        serverShutdownMessageBox->close();
+    serverShutdownMessageBox = new QMessageBox(this);
```

Comment:
Creating a `new` so that we pass in the parent. Passing in the parent will display the root icon in the corner of the popup without the need to set the icon manually each time. I have set an attribute now that will delete the widget on closing: `serverShutdownMessageBox->setAttribute(Qt::WA_DeleteOnClose);`